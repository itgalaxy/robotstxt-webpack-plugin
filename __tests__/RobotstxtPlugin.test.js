const { webpack } = require("./helpers");

describe("plugin", () => {
  it("should execute successfully", async () => {
    const stats = await webpack("entry.js");

    const { warnings, errors, assets } = stats.compilation;

    expect(warnings).toMatchSnapshot("warnings");
    expect(errors).toMatchSnapshot("errors");
    expect(assets["robots.txt"].source()).toMatchSnapshot();
  });

  it("should execute successfully with `filePath` option", async () => {
    const stats = await webpack("entry.js", {
      pluginOptions: {
        filePath: "robots.dev.txt"
      }
    });

    const { warnings, errors, assets } = stats.compilation;

    expect(warnings).toMatchSnapshot("warnings");
    expect(errors).toMatchSnapshot("errors");
    expect(assets["robots.dev.txt"].source()).toMatchSnapshot();
  });

  it("should throw error on invalid `generate-robotstxt` options", async () => {
    const stats = await webpack("entry.js", {
      pluginOptions: {
        policy: {}
      }
    });

    const { warnings, errors } = stats.compilation;

    expect(warnings).toMatchSnapshot("warnings");
    expect(errors).toMatchSnapshot("errors");
  });
});
