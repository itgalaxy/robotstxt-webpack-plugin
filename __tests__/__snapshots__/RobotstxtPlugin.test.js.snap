// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`plugin should execute successfully 1`] = `
"User-agent: *
Allow: /
"
`;

exports[`plugin should execute successfully with \`filePath\` option 1`] = `
"User-agent: *
Allow: /
"
`;

exports[`plugin should execute successfully with \`filePath\` option: errors 1`] = `Array []`;

exports[`plugin should execute successfully with \`filePath\` option: warnings 1`] = `Array []`;

exports[`plugin should execute successfully: errors 1`] = `Array []`;

exports[`plugin should execute successfully: warnings 1`] = `Array []`;

exports[`plugin should throw error on invalid \`generate-robotstxt\` options: errors 1`] = `
Array [
  [Error: Options \`policy\` must be array],
]
`;

exports[`plugin should throw error on invalid \`generate-robotstxt\` options: warnings 1`] = `Array []`;
